maintenance/projects/senna/src/screen/RequestScreen.js

```diff
@@ -274,8 +274,9 @@ class RequestScreen extends Screen {
 						reason.requestError = true;
 						reason.requestPrematureTermination = true;
 						break;
+					default:
+						throw reason;
 				}
-				throw reason;
```

Comment:
This seems like slightly different logic. Maybe I am not reading this right but I am seeing that the `throw` 

Previous Code:
- Run switch statement and set timeout/requestError
- Then, `throw reason`

New Code:
- Run switch statement and set timeout/requestError
- OR `throw reason`.

With the new code, we won't throw unless one of the `case`'s are not met

Reply:
:open_mouth: Good catch